### ISSUE TITLE
Tools request - chloroplast and mitochondrial assemblies

### ISSUE THREAD

**Issue by dougsd3** [galaxyproject/tools-iuc#3392]
One thing thet really called my attention was the lack or chloroplast/mitochondrial assembly tools in Galaxy. Recently, Novoplasty was included, but I feel others could be added.

I would recommend:

1- Getorganelle: https://github.com/Kinggerm/GetOrganelle (assembles chloroplast and mitochondria)
2- Organelle_PBA: https://github.com/aubombarely/Organelle_PBA (devoted to PacBio reads)
3- Chloro: https://github.com/sestaton/Chloro (an abandoned project who might be adopted by Galaxy)
4- MITObin (devoted for mitochondria, but I've heard it works well with chloroplasts): https://github.com/chrishah/MITObim https://github.com/galaxyproject/tools-iuc/pull/3400
4- OGA (Organelle Genome assembler): https://github.com/quxiaojian/OGA

These requests are quite in order of priority I would recommend.

best regards
Douglas

**Comment by bernt-matthias** [comment]
Is there an order in the list? Do you have any favorites?

**Comment by bernt-matthias** [comment]
Could you check which are allready on bioconda?

**Comment by dougsd3** [comment]
I put them pretty much in the order.

I'll if they're available at bioconda.

Thanks
Douglas


Em dom, 20 de dez de 2020 13:47, M Bernt <notifications@github.com>
escreveu:

**Comment by dougsd3** [comment]
Getorganelle is the only one avaliable in bioconda, but it is the most up-to-date one. I would strongly recommend going further with it, because it fits to mitochondrial and chloroplast genomes

**Comment by dougsd3** [comment]
https://bioconda.github.io/recipes/getorganelle/README.html

**Comment by bernt-matthias** [comment]
@dougsd3 even if not your favourite you may want to have a look here https://github.com/galaxyproject/tools-iuc/pull/3400 .. comments welcome. 

For GetOrganelle some changes might be necessary https://github.com/Kinggerm/GetOrganelle/issues/64

**Comment by bernt-matthias** [comment]
Maybe the list should be extended by 

- https://github.com/ndierckx/NOVOPlasty
- https://github.com/ibest/ARC
- https://git.metabarcoding.org/org-asm/org-asm

**Comment by dougsd3** [comment]
that's good
-- 
Douglas Silva Domingues, PhD
Research Professor
Group of Genomics and Transcriptomes in Plants
Institute of Biosciences at Rio Claro
UNESP - Sao Paulo State University
Av. 24-A, 1515 - Botany Building
13506-900, Rio Claro, SP, Brazil
Phone: +55 19 35264207
e-mail: doug@rc.unesp.br
GoogleScholar: http://tinyurl.com/dougpublications
<https://tinyurl.com/dougpublications>


Em ter., 29 de dez. de 2020 às 17:07, M Bernt <notifications@github.com>
escreveu:

**Comment by stephanflemming** [comment]
plasmidSPAdes is already part of this PR https://github.com/galaxyproject/tools-iuc/pull/2612

**Comment by bernt-matthias** [comment]
https://github.com/pedronachtigall/MITGARD

**Comment by dougsd3** [comment]
Good to know.
I put Pedro in cc.

Hi Pedro, people from Galaxy are considering incorporating your tool to
their server.

Em qui., 4 de fev. de 2021 às 07:44, M Bernt <notifications@github.com>
escreveu:

**Comment by pedronachtigall** [comment]
Hi,

Sorry for the slow response, I had a really busy week.

Thank you @bernt-matthias for considering MITGARD to be integrated into the Galaxy platform.
Thank you @dougsd3 for cc me about this issue.

What should we need to make it happen?

Best,
Pedro

**Comment by bernt-matthias** [comment]
Hi @pedronachtigall 

> What should we need to make it happen?

- MITGARD needs to be added to bioconda.
- Then the tool wrapper needs to be created. Question is if you would like to do this on your own? I always consider it the best way if the developer does this .. we can certainly support you.
- A quick look on your github page showed me that you use the ete3 module to download taxonomy. If I remember correctly this is then stored in the home directory. This won't work, i.e. for Galaxy this would need to be configurable. We have a data manager that downloads the taxdump from the NCBI ftp. Could MITGARD be congigured to work on such a taxdump?
- Are there any other databases involved?

**Comment by pedronachtigall** [comment]
Hi @bernt-matthias 

MITGARD needs to be added to bioconda.
- I will work to get MITGARD added to bioconda and I will ask for support if needed. Just to know, do you have any easy guide (or step-by-step guide) to do that?

A quick look on your github page showed me that you use the ete3 module to download taxonomy. If I remember correctly this is then stored in the home directory. This won't work, i.e. for Galaxy this would need to be configurable. We have a data manager that downloads the taxdump from the NCBI ftp. Could MITGARD be congigured to work on such a taxdump?
- You are right, the taxonomy downloaded from ete3 module is saved in the home directory. One important thing is that MITGARD did not use the ete3 module directly. It is used by MitoZ, which is one of the assemblers incorporated into MITGARD. MitoZ did not perform so well with RNA-seq data, it was just implemented to help confirm some regions. Another point is that MitoZ is not able to be installed through conda, it needs to be set up manually (or used with Docker or Singularity container). So, this assembler must be a problem for the conda installation.
    - What should we do about this issue?
    - I can implement MITGARD with an option to turn on/off the MitoZ software and make it off by default. Then, Galaxy may not allow the user to set this option on. What do you think?

Are there any other databases involved?
- No, there are no other databases involved.

**Comment by bgruening** [comment]
@pedronachtigall for Bioconda we have here some documentation: https://bioconda.github.io/contributor/index.html let me know if you need help.

**Comment by bernt-matthias** [comment]
> I can implement MITGARD with an option to turn on/off the MitoZ software and make it off by default. Then, Galaxy may not allow the user to set this option on. What do you think?

Yep, this is possible (you do not need to disable MitoZ by default .. but this is up to you).

But it seems possible to add MitoZ to bioconda as well (someone already tried https://github.com/bioconda/bioconda-recipes/pull/22334). They have (kind of) releases: https://github.com/linzhi2013/MitoZ/blob/master/version_2.3/release_MitoZ_v2.3.tar.bz2 which could be used. Lets see if we can reactivate this. Which version of MitoZ would you need? 

But we can go without MitoZ it the first place and readd it later...

**Comment by bernt-matthias** [comment]
- Mitomaker: https://www.preprints.org/manuscript/201808.0423/v1
- Mitofinder: https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13160

**Comment by bgruening** [comment]
We are currently working on `mitohifi` which is using mitofinder. Mitofinder has a few problems to deploy it in conda, so we have a Docker image for it, which we also use for mitohifi.

**Comment by dougsd3** [comment]
Great news

Thanks !!!

Em qui, 27 de mai de 2021 06:17, Björn Grüning ***@***.***>
escreveu:

**Comment by pedronachtigall** [comment]
Hi @bernt-matthias and @bgruening ,

After a long time, I had a chance to work to get MITGARD as a docker repository and modify MITGARD to have MitoZ as an option.
Now MITGARD can be used through docker `docker pull pedronachtigall/mitgard:latest`. The Dockerfile can be also found in MITGARD repository. Is it enough to implement MITGARD into Galaxy?
If we still need to make it available through conda, I already have a meta.yaml with all dependencies (except for MitoZ, which is optional now.

Can you help me to design a build.sh file to have it installed through conda?
https://github.com/pedronachtigall/MITGARD

I will appreciate your help.

Best regards,
Pedro

**Comment by bernt-matthias** [comment]
Wonderful @pedronachtigall . `conda` would be super convenient for Galaxy. Seems that the build file only needs to copy the python scripts to `${PREFIX}/bin`. Also you will need a release on your github project. 

You may just open a PR (make it a draft) on bioconda recipes and ping me.

Btw, at the training week of the Galaxy community conference (end of June) there is a tutorial explaining how to get a running Galaxy tool starting with conda, over tool building and testing ...

**Comment by pedronachtigall** [comment]
Hi @bernt-matthias , I already release the first version of MITGARD and upload build.sh and meta.yaml files into recipes of bioconda repository. I pig you into the PR. Hope it works fine.

FYI, I also take the opportunity to add a PR for another tool named CodAn (https://github.com/pedronachtigall/CodAn), which I also want to have it easily installed through conda.

Let me know if everything is fine, or if I need to change something in the files.

**Comment by pedronachtigall** [comment]
Hi, MITGARD is ready to go through Conda installation: https://anaconda.org/bioconda/mitgard
Just let me know if you need anything else to have it integrated into the Galaxy.

**Comment by pedronachtigall** [comment]
Hi, do you have any news about adding MITGARD into Galaxy? Is the tutorial available (the tutorial mentioned in the "training week of the Galaxy community")?

**Comment by o-william-white** [comment]
@bernt-matthias I have seen that GetOrganelle has been requested for galaxy and some work has already been done https://github.com/Kinggerm/GetOrganelle/issues/64 

Just wondering it is going to be added to Galalxy toolshed? Apologies if I have missed it

**Comment by bernt-matthias** [comment]
Hi @o-william-white unfortulately I have not found time to work on this. But I'm happy to help if someone wants to jump in and submit a tool wrapper.

**Comment by o-william-white** [comment]
Hi @bernt-matthias, thank you for letting me know, I've been having a look at the documentation about how to set up a tool wrapper. If I make any progress I will let you know.

**Comment by bernt-matthias** [comment]
Hi @o-william-white wonderful. Maye just open a PR to the IUC with the current state. Then we can give some suggestions.

**Comment by o-william-white** [comment]
Hi @bernt-matthias thanks for the reply. I just wanted to check a few things to make sure I understand as I havent done this before. A PR is a pull resquest? To do this, I create a fork of the tools-iuc repository, add my xml wrapper to the tools folder and then create a pull request. Presumably this pull request isn't intended to be a finished product, but it will give you the oppertunity to suggest changes? Apologies for checking, just wanted to make sure I didn't do something annoying!

**Comment by bernt-matthias** [comment]
@o-william-white exactly. Don't forget the `.shed.yaml` file, otherwise tests won't run.

**Comment by o-william-white** [comment]
@bernt-matthias awesome, thank you. I will use a .shed.yaml file from another tool as a template and hopefully that works

**Comment by o-william-white** [comment]
@bernt-matthias I have added the PR https://github.com/galaxyproject/tools-iuc/pull/4455 hopefully this is all ok! Thanks for the advice
